modules/library.js

```diff
@@ -0,0 +1,60 @@
+class Library {
+  constructor() {
+    this.bookData = [];
+    this.bookData = JSON.parse(localStorage.getItem('bookstoredata'));
+    this.formInput = document.getElementById('book-form');
+  }
+
+  showSavedData = () => {
+    if (this.bookData) {
+      let bookList = '';
+      if (this.bookData.length > 0) {
+        bookList = '<div class="table-container">';
+      } else {
+        bookList = '<div>';
+      }
+
+      bookList += '<table class="table table-hover table-sm table-striped"><tbody>';
+      this.bookData.forEach((item, i) => {
+        bookList += `        
+        <tr>
+          <td>"${item.title}" by ${item.author}</td>
+          <td><button type="submit" class="remove-button" id="${i}">Remove</button></td>
+        </tr>          
+        `;
+      });
+      bookList += '</tbody></table></div>';
+      const bookCollection = document.getElementById('book-collection');
+      bookCollection.innerHTML = bookList;
+      const removeButtons = document.querySelectorAll('.remove-button');
+      removeButtons.forEach((item) => item.addEventListener('click', this.removeBook.bind(this)));
+    }
+  }
+
+  addBookData = (event) => {
+    event.preventDefault();
+    const title = document.getElementById('title');
+    const author = document.getElementById('author');
+    if (title.value.length > 0 && author.value.length > 0) {
+      this.bookData.push({
+        title: title.value,
+        author: author.value,
+      });
+      this.storeData();
+      this.showSavedData();
+      this.formInput.reset();
+    }
+  }
```

Comment:
- [ ] Great job refactoring your code. However, the web app should function normally. The user is not able to add books. I tried to add a book, but I could not find it on the List page, not even in the local storage. Please make sure the book is displayed on the List page and saved to the local storage.

Note: Once you resolve the issue with adding books, please double-check that deleting books works properly before submitting a new review.

Reply:
Hi @medaminedev66, thanks for the valuable feedback. But I would like to know whether you tested it by running Live Server for VS code or without running the server. As this code uses different modules we need to run a server like Live Server for VS Code as the Chrome browser by itself has issues with CORS policy. In my system when I test it by running the Live server, I can add, delete and see the books in the list. So, could you please recheck and let me know?